_opensearch/geohexgrid-agg.md

```diff
@@ -0,0 +1,377 @@
+  },
+  "hits" : {
+    "total" : {
+      "value" : 3,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "national_parks",
+        "_id" : "1",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Yellowstone National Park",
+          "location" : "44.42, -110.59"
+        }
+      },
+      {
+        "_index" : "national_parks",
+        "_id" : "2",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Yosemite National Park",
+          "location" : "37.87, -119.53"
+        }
+      },
+      {
+        "_index" : "national_parks",
+        "_id" : "3",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Death Valley National Park",
+          "location" : "36.53, -116.93"
+        }
+      }
+    ]
+  },
+  "aggregations" : {
+    "grouped" : {
+      "buckets" : [
+        {
+          "key" : "8129bffffffffff",
+          "doc_count" : 2
+        },
+        {
+          "key" : "8128bffffffffff",
+          "doc_count" : 1
+        }
+      ]
+    }
+  }
+}
+```
+
+## High-precision requests
+
+Now run a higher precision request:
+
+```json
+GET national_parks/_search
+{
+  "aggregations": {
+    "grouped": {
+      "geohex_grid": {
+        "field": "location",
+        "precision": 6
+      }
+    }
+  }
+}
+```
+
+All three documents are bucketed separately because of higher granularity:
+
+```json
+{
+  "took" : 5,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 3,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "national_parks",
+        "_id" : "1",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Yellowstone National Park",
+          "location" : "44.42, -110.59"
+        }
+      },
+      {
+        "_index" : "national_parks",
+        "_id" : "2",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Yosemite National Park",
+          "location" : "37.87, -119.53"
+        }
+      },
+      {
+        "_index" : "national_parks",
+        "_id" : "3",
+        "_score" : 1.0,
+        "_source" : {
+          "name" : "Death Valley National Park",
+          "location" : "36.53, -116.93"
+        }
+      }
+    ]
+  },
+  "aggregations" : {
+    "grouped" : {
+      "buckets" : [
+        {
+          "key" : "8629ab6dfffffff",
+          "doc_count" : 1
+        },
+        {
+          "key" : "8629857a7ffffff",
+          "doc_count" : 1
+        },
+        {
+          "key" : "862896017ffffff",
+          "doc_count" : 1
+        }
+      ]
+    }
+  }
+}
+```
+
+## Filtering requests
+
+High-precision requests are resource-intensive, so we recommend using a filter like `geo_bounding_box` to limit the geographical area. For example, the following query applies a filter to limit the search area:
```

Comment:
Compound modifiers only need to be hyphenated when they precede the noun they modify. So it's a resource-intensive request, but the request is resource intensive.